### PR TITLE
new classes

### PR DESCRIPTION
**Pull request by redmarverick**
### New classes created

- Create a class Nameable. ✔️
- Implement a method called correct_name that will raise a NotImplementedError. ✔️
- Turn your Person class to Nameable ✔️
- Make sure that your Person class inherits from Nameable ✔️
- Make sure that this class has a method correct_name implemented. It should simply return the name attribute. ✔️
- Make sure that it inherits from Nameable. ✔️
- In the constructor assign a nameable object from params to an instance variable. ✔️
- Implement the correct_name method that returns the result of the correct_name method of the @nameable. ✔️
- Prepare CapitalizeDecorator and TrimmerDecorator:
- Create a class that inherits from the base Decorator class. ✔️
- Implement a method correct_name that capitalizes the output of @nameable.correct_name. ✔️
- Create a class that inherits from the base Decorator class. ✔️
- Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 - characters. If it's longer it should trim the word. ✔️
- See your decorators in action ✔️

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -22,4 +20,8 @@ def generate_id
   def can_use_services?
     true
   end
+
+  def correct_name
+    @name
+  end
```
Comment (thecodechaser) [inline person.rb]: - [x] Please make sure the `Person` class is inherited from the `Namebale` class for the associations to work as expected.
  - You can consider the code `class Person < Nameable`

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require_relative 'nameable'
+
+class Decorator
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
```
Comment (thecodechaser) [inline decorator.rb]: - [x] Please make sure the `Decorator` class is inherited from the `Namebale` class for the associations to work as expected.
  - You can consider the code `class Decorator < Nameable`

Comment (thecodechaser) [inline decorator.rb]: - [x] Please make sure the `nameable` variable is readable and writable for the associations to work as expected.
  - You can consider the code `attr_accessor :nameable`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @redmarverick👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] The LIcense link in the README file is broken, Please make sure to fix this. It will make the README file more professional.

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — CHANGES_REQUESTED** [review]
Hi @redmarverick ,

Great job so far! 

May I highlight the following: 
- [x] Your project is professionally documented👏  ✔️
- [x] Well done implementing most of the requirements👏  ✔️



## CHANGES REQUIRED:muscle::clap:

- [ REQUIRED] Kindly as required by the previous TSE fix the broken link of the MIT license in the README.

Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Review by TadesseDev — APPROVED** [review]
### Hi @redmarverick 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**